packages/block-library/src/spacer/controls.js

```diff
@@ -41,32 +40,27 @@ function DimensionInput( { label, onChange, isResizing, value = '' } ) {
 	} );
 
 	const handleOnChange = ( unprocessedValue ) => {
-		setTemporaryInput( null );
 		onChange( unprocessedValue );
 	};
 
-	const handleOnBlur = () => {
-		if ( temporaryInput !== null ) {
-			setTemporaryInput( null );
-		}
-	};
-
-	const inputValue = temporaryInput !== null ? temporaryInput : value;
+	const computedValue = useMemo( () => {
+		const [ parsedQuantity, parsedUnit ] = parseQuantityAndUnitFromRawValue(
+			value
+		);
+		// Force the unit to update to `px` when the Spacer is being resized.
+		return [ parsedQuantity, isResizing ? 'px' : parsedUnit ].join( '' );
+	}, [ isResizing, value ] );
 
 	return (
 		<BaseControl label={ label } id={ inputId }>
 			<UnitControl
 				id={ inputId }
-				isResetValueOnUnitChange
```

Comment:
I think it might be better to keep this line, so that we can preserve the defaults when switching between units. Otherwise switching from `px` to `em` or `rem` can be a bit overwhelming in the editor

![spacer-switch-units](https://user-images.githubusercontent.com/14988353/158916125-ad606902-431c-4e98-8287-d7b48eba47e2.gif)

Reply:
Agreed, I must have overlooked the consequences of removing this prop.

Added it back in https://github.com/WordPress/gutenberg/pull/39513/commits/fd18928678a55b369dc2ec8c8e07dea646293fde